mcs/class/corlib/System/TimeZoneInfo.Unity.cs

```diff
@@ -61,7 +61,9 @@ static List<AdjustmentRule> CreateAdjustmentRule (int year, out Int64[] data, ou
 			// If the first and second transition DateTime objects are the same, ValidateAdjustmentRule will throw
 			// an exception. I'm unsure why these would be the same, but we do see that occur for some locales.
 			// In that case, just exit early.
-			if (firstTransition.Equals(secondTransition))
+			if (firstTransition.TimeOfDay.Equals(secondTransition.TimeOfDay)
+				&& firstTransition.Month.Equals(secondTransition.Month)
+				&& firstTransition.Day.Equals(secondTransition.Day))
 				return rulesForYear;
```

Comment:
We should add to the comment that the years can be different. When I read this without intellisense I ask what parts can be different and why